### ISSUE TITLE
build failure after 23.3.3 upgrade

### ISSUE THREAD

**Issue by triaxx** [processone/ejabberd#3594]
## Environment

- ejabberd version: 21.04
- Erlang version: Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 11.1.8
- OS: NetBSD 9.0
- Installed from: source

## Bug description

```erlang
Compiled src/ejabberd_http_ws.erl
Compiled src/ejabberd_db_sup.erl
Compiled src/ejabberd_doc.erl
ejabberd_http: function extract_path_query/1+74:
  Internal consistency check failed - please report this bug.
  Instruction: {call,1,{f,302}}
  Error:       {bad_arg_type,
                   {x,0},
                   {t_tuple,22,true,
                       #{1 => {t_atom,[state]},
                         4 => {t_atom,['POST','PUT']},
                         6 => {t_tuple,2,true,#{1 => {t_atom,[abs_path]}}}}},
                   {t_tuple,22,true,
                       #{1 => {t_atom,[state]},4 => {t_atom,['POST']}}}}:

Compiling src/ejabberd_http.erl failed:
ERROR: compile failed while processing /mnt/ccd0/pkgsrc/work/netbsd-9-unprivileged/chat/ejabberd/work/ejabberd-21.04: rebar_abort                                                                                                             
gmake: *** [Makefile:122: src] Error 1
*** Error code 2

Stop.
```

**Comment by prefiks** [comment]
Hello,

This message is about problem/bug in compiler, as i understand this code compiles correctly on 23.3.2?

I don't think there is anything wrong with that function, at least i don't see anything tricky or something that is invalid per erlang rules.

**Comment by prefiks** [comment]
Ok i can reproduce this with this version, i believe this is same bug as reported in https://github.com/erlang/otp/issues/4813 in erlang repository

**Comment by prefiks** [comment]
This bug should be fixed on OTP-23.3.4 that was just released.
